src/Components/Facility/Consultations/Beds.tsx

```diff
@@ -52,33 +48,30 @@ const Beds = (props: BedsProps) => {
   const [key, setKey] = useState(0);
   const [showBedDetails, setShowBedDetails] = useState<CurrentBed | null>(null);
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
-      setIsLoading(true);
-      const [bedsData]: any = await Promise.all([
-        dispatch(listConsultationBeds({ consultation: consultationId })),
-      ]);
-      if (!status.aborted) {
-        setIsLoading(false);
-        if (!bedsData?.data)
-          Notification.Error({
-            msg: "Something went wrong..!",
-          });
-        else {
-          setConsultationBeds(bedsData.data.results);
-          setBed(bedsData.data.results[0]?.bed_object || {});
-          setAssets(bedsData.data.results[0]?.assets_objects || []);
-        }
-      }
-    },
-    [consultationId, dispatch]
-  );
-  useAbortableEffect(
-    (status: statusType) => {
-      fetchData(status);
-    },
-    [dispatch, fetchData, key]
-  );
+  const {
+    res,
+    data: bedData,
+    refetch,
+  } = useQuery(routes.listConsultationBeds, {
+    pathParams: { consultation: consultationId },
+  });
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (!bedData || !res?.ok) {
+      Notification.Error({
+        msg: "Something went wrong..!",
+      });
+    } else {
+      setConsultationBeds(bedData.results);
```

Comment:
Again, why have an additional state when `useQuery` already maintains a state for you?

Remove the consultationBeds state and use the `data?.results` directly instead.